bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtoEncoding.java

```diff
@@ -356,7 +355,7 @@ private static ByteBuf serializeProtobuf(MessageLite msg, ByteBufAllocator alloc
         ByteBuf buf = allocator.heapBuffer(size, size);
 
         try {
-            msg.writeTo(CodedOutputStream.newInstance(buf.array(), buf.arrayOffset() + buf.writerIndex(), size));
+            msg.writeTo(new ByteBufOutputStream(buf));
```

Comment:
Is this a race or corruption happens all the time?  Change looks fine, but I did not understand the actual issue with the current code on why corruption happening.

Reply:
@jvrao there is no conclusion yet. this is just marked as WIP.